### PR TITLE
determine OCI-SIF vs. native SIF using image contents, not URI

### LINKED ISSUE #2184: remove brute-force oci-sif vs. native sif determination...?
It seems like the code we now have in `pkg/image` satisfies the precondition in this TODO:

https://github.com/sylabs/singularity/blob/76ccdddf6323fd11c9444c8251d3474570c93ced/cmd/internal/cli/actions.go#L207C8-L207C8

```golang

	// TODO - drop once we have implemented prefix-less oci-sif vs native sif detection.
	if isOCI {
		image = "oci-sif:" + image
	}

```

Is that correct? And if so, is it time to address this?

### PR DESCRIPTION
**Pull request by preminger**
## Description of the Pull Request (PR):

The code in `pkg/image` is now able to detect OCI-SIF images (and distinguish them from native SIFs) based on image content. This PR replaces reliance on the URI transport field being `oci-sif:` with detection of OCI-SIF images based on the image content.

### This fixes or addresses the following GitHub issues:

 - Fixes #2184

### CODE REVIEW

File: cmd/internal/cli/actions.go [hunk 1]
```diff
@@ -200,15 +201,6 @@ func replaceURIWithImage(ctx context.Context, cmd *cobra.Command, args []string)
 		return origImageURI
 	}
 
-	if err != nil {
```
Comment (dtrudg) [inline cmd/internal/cli/actions.go:203]: Don't think this should be dropped?

It is handling where one of the `case`s returns an error which is not a OCI-SIF mount issue.

File: internal/pkg/util/image/ocisif.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// Copyright (c) 2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package image
```
Comment (dtrudg) [inline internal/pkg/util/image/ocisif.go:6]: I don't see a good reason not to have this as a public convenience function in `pkg/image`?

I don't think it is at risk of an API change.

**Review by dtrudg — CHANGES_REQUESTED** [review]

File: cmd/internal/cli/actions.go (hunk 1 above)
Comment (preminger) [reply cmd/internal/cli/actions.go:203]: Yeah, not sure how that ended up being part of what I removed...

File: pkg/image/ocisif.go [hunk 3]
```diff
@@ -60,3 +61,13 @@ func (f *ociSifFormat) openMode(bool) int {
 func (f *ociSifFormat) lock(*Image) error {
 	return nil
 }
+
+func IsOCISIF(filename string) (bool, error) {
```
Comment (dtrudg) [inline pkg/image/ocisif.go:65]: As this is an importable pkg, there should be a docstring on the function.

**Review by dtrudg — APPROVED** [review]
LGTM - but please add a doc string on the function in `pkg`.

**Pull request merged**